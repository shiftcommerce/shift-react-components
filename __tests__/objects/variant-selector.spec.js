// Libraries
import React from 'react'

// Object
import VariantSelector from '../../src/objects/variant-selector'

// Fixtures
import product from '../fixtures/product'

test('renders correctly', () => {
  // Arrange
  const changeVariant = jest.fn()

  // Act
  const wrapper = mount(
    <VariantSelector name='line_item[item_id]' prompt='Select a Size' onClick={changeVariant} variants={product.variants} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  wrapper.find('div.o-variant-selector').simulate('click', { target: { value: 'Variant 1' } })
  expect(changeVariant).toHaveBeenCalled()
})

test('renders a button for each variant', () => {
  // Arrange
  const changeVariant = jest.fn()

  // Act
  const wrapper = mount(
    <VariantSelector name='line_item[item_id]' prompt='Select a Size' onClick={changeVariant} variants={product.variants} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('button.c-product-display__option-button')).toHaveLength(product.variants.length)
})
