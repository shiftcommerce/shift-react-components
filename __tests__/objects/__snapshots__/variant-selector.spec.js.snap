// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button for each variant 1`] = `
<VariantSelector
  name="line_item[item_id]"
  onClick={[MockFunction]}
  prompt="Select a Size"
  variants={
    Array [
      Object {
        "available_to_browse": false,
        "available_to_order": false,
        "created_at": "2018-09-24T15:43:05Z",
        "current_price": 16.95,
        "deleted_at": null,
        "description": "Product 51 - variant 1

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
        "ewis_eligible": false,
        "expected_availability_date": null,
        "has_assets": true,
        "id": "343",
        "meta_attributes": Object {},
        "ordered_total": 10,
        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
        "position": null,
        "price": 16.95,
        "price_includes_taxes": true,
        "product_id": 175,
        "reference": "dolor_50_1",
        "sku": "9337040128595",
        "stock_allocated_level": 10,
        "stock_available_level": 1,
        "stock_level": 11,
        "tax_code": "vat",
        "title": "Variant 1",
        "updated_at": "2018-11-05T11:23:16Z",
      },
      Object {
        "available_to_browse": false,
        "available_to_order": false,
        "created_at": "2018-09-24T15:43:05Z",
        "current_price": 97.68,
        "deleted_at": null,
        "description": "Product 51 - variant 2

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
        "ewis_eligible": false,
        "expected_availability_date": null,
        "has_assets": true,
        "id": "344",
        "meta_attributes": Object {},
        "ordered_total": 1,
        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
        "position": null,
        "price": 97.68,
        "price_includes_taxes": true,
        "product_id": 175,
        "reference": "voluptas_50_2",
        "sku": "0927967530244",
        "stock_allocated_level": 1,
        "stock_available_level": 0,
        "stock_level": 1,
        "tax_code": "vat",
        "title": "Variant 2",
        "updated_at": "2018-10-31T10:24:00Z",
      },
      Object {
        "available_to_browse": false,
        "available_to_order": false,
        "created_at": "2018-09-24T15:43:05Z",
        "current_price": 97.68,
        "deleted_at": null,
        "description": "Product 51 - variant 2

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
        "ewis_eligible": true,
        "expected_availability_date": null,
        "has_assets": true,
        "id": "345",
        "meta_attributes": Object {},
        "ordered_total": 1,
        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
        "position": null,
        "price": 97.68,
        "price_includes_taxes": true,
        "product_id": 175,
        "reference": "voluptas_50_3",
        "sku": "0927967530245",
        "stock_allocated_level": 1,
        "stock_available_level": 0,
        "stock_level": 1,
        "tax_code": "vat",
        "title": "Variant 3",
        "updated_at": "2018-10-31T10:24:00Z",
      },
    ]
  }
>
  <div
    className="o-variant-selector"
    name="line_item[item_id]"
    onClick={[MockFunction]}
  >
    <Button
      aria-posinset={1}
      aria-setsize={3}
      className="c-product-display__option-button "
      data-price={16.95}
      data-stock-available-level={1}
      data-variant-id="343"
      key="343"
      label="Variant 1 - £16.95"
      onClick={[Function]}
      type="button"
      value="9337040128595"
    >
      <button
        aria-posinset={1}
        aria-setsize={3}
        className="c-product-display__option-button  o-button"
        data-price={16.95}
        data-stock-available-level={1}
        data-variant-id="343"
        disabled={false}
        onClick={[Function]}
        role="button"
        tabIndex="0"
        type="button"
        value="9337040128595"
      >
        Variant 1 - £16.95
      </button>
    </Button>
    <Button
      aria-posinset={2}
      aria-setsize={3}
      className="c-product-display__option-button "
      data-price={97.68}
      data-stock-available-level={0}
      data-variant-id="344"
      key="344"
      label="Variant 2 - £97.68- Out of Stock"
      onClick={[Function]}
      type="button"
      value="0927967530244"
    >
      <button
        aria-posinset={2}
        aria-setsize={3}
        className="c-product-display__option-button  o-button"
        data-price={97.68}
        data-stock-available-level={0}
        data-variant-id="344"
        disabled={false}
        onClick={[Function]}
        role="button"
        tabIndex="0"
        type="button"
        value="0927967530244"
      >
        Variant 2 - £97.68- Out of Stock
      </button>
    </Button>
    <Button
      aria-posinset={3}
      aria-setsize={3}
      className="c-product-display__option-button "
      data-price={97.68}
      data-stock-available-level={0}
      data-variant-id="345"
      key="345"
      label="Variant 3 - £97.68- Email When in Stock"
      onClick={[Function]}
      type="button"
      value="0927967530245"
    >
      <button
        aria-posinset={3}
        aria-setsize={3}
        className="c-product-display__option-button  o-button"
        data-price={97.68}
        data-stock-available-level={0}
        data-variant-id="345"
        disabled={false}
        onClick={[Function]}
        role="button"
        tabIndex="0"
        type="button"
        value="0927967530245"
      >
        Variant 3 - £97.68- Email When in Stock
      </button>
    </Button>
  </div>
</VariantSelector>
`;

exports[`renders correctly 1`] = `
<VariantSelector
  name="line_item[item_id]"
  onClick={[MockFunction]}
  prompt="Select a Size"
  variants={
    Array [
      Object {
        "available_to_browse": false,
        "available_to_order": false,
        "created_at": "2018-09-24T15:43:05Z",
        "current_price": 16.95,
        "deleted_at": null,
        "description": "Product 51 - variant 1

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
        "ewis_eligible": false,
        "expected_availability_date": null,
        "has_assets": true,
        "id": "343",
        "meta_attributes": Object {},
        "ordered_total": 10,
        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
        "position": null,
        "price": 16.95,
        "price_includes_taxes": true,
        "product_id": 175,
        "reference": "dolor_50_1",
        "sku": "9337040128595",
        "stock_allocated_level": 10,
        "stock_available_level": 1,
        "stock_level": 11,
        "tax_code": "vat",
        "title": "Variant 1",
        "updated_at": "2018-11-05T11:23:16Z",
      },
      Object {
        "available_to_browse": false,
        "available_to_order": false,
        "created_at": "2018-09-24T15:43:05Z",
        "current_price": 97.68,
        "deleted_at": null,
        "description": "Product 51 - variant 2

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
        "ewis_eligible": false,
        "expected_availability_date": null,
        "has_assets": true,
        "id": "344",
        "meta_attributes": Object {},
        "ordered_total": 1,
        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
        "position": null,
        "price": 97.68,
        "price_includes_taxes": true,
        "product_id": 175,
        "reference": "voluptas_50_2",
        "sku": "0927967530244",
        "stock_allocated_level": 1,
        "stock_available_level": 0,
        "stock_level": 1,
        "tax_code": "vat",
        "title": "Variant 2",
        "updated_at": "2018-10-31T10:24:00Z",
      },
      Object {
        "available_to_browse": false,
        "available_to_order": false,
        "created_at": "2018-09-24T15:43:05Z",
        "current_price": 97.68,
        "deleted_at": null,
        "description": "Product 51 - variant 2

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
        "ewis_eligible": true,
        "expected_availability_date": null,
        "has_assets": true,
        "id": "345",
        "meta_attributes": Object {},
        "ordered_total": 1,
        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
        "position": null,
        "price": 97.68,
        "price_includes_taxes": true,
        "product_id": 175,
        "reference": "voluptas_50_3",
        "sku": "0927967530245",
        "stock_allocated_level": 1,
        "stock_available_level": 0,
        "stock_level": 1,
        "tax_code": "vat",
        "title": "Variant 3",
        "updated_at": "2018-10-31T10:24:00Z",
      },
    ]
  }
>
  <div
    className="o-variant-selector"
    name="line_item[item_id]"
    onClick={[MockFunction]}
  >
    <Button
      aria-posinset={1}
      aria-setsize={3}
      className="c-product-display__option-button "
      data-price={16.95}
      data-stock-available-level={1}
      data-variant-id="343"
      key="343"
      label="Variant 1 - £16.95"
      onClick={[Function]}
      type="button"
      value="9337040128595"
    >
      <button
        aria-posinset={1}
        aria-setsize={3}
        className="c-product-display__option-button  o-button"
        data-price={16.95}
        data-stock-available-level={1}
        data-variant-id="343"
        disabled={false}
        onClick={[Function]}
        role="button"
        tabIndex="0"
        type="button"
        value="9337040128595"
      >
        Variant 1 - £16.95
      </button>
    </Button>
    <Button
      aria-posinset={2}
      aria-setsize={3}
      className="c-product-display__option-button "
      data-price={97.68}
      data-stock-available-level={0}
      data-variant-id="344"
      key="344"
      label="Variant 2 - £97.68- Out of Stock"
      onClick={[Function]}
      type="button"
      value="0927967530244"
    >
      <button
        aria-posinset={2}
        aria-setsize={3}
        className="c-product-display__option-button  o-button"
        data-price={97.68}
        data-stock-available-level={0}
        data-variant-id="344"
        disabled={false}
        onClick={[Function]}
        role="button"
        tabIndex="0"
        type="button"
        value="0927967530244"
      >
        Variant 2 - £97.68- Out of Stock
      </button>
    </Button>
    <Button
      aria-posinset={3}
      aria-setsize={3}
      className="c-product-display__option-button "
      data-price={97.68}
      data-stock-available-level={0}
      data-variant-id="345"
      key="345"
      label="Variant 3 - £97.68- Email When in Stock"
      onClick={[Function]}
      type="button"
      value="0927967530245"
    >
      <button
        aria-posinset={3}
        aria-setsize={3}
        className="c-product-display__option-button  o-button"
        data-price={97.68}
        data-stock-available-level={0}
        data-variant-id="345"
        disabled={false}
        onClick={[Function]}
        role="button"
        tabIndex="0"
        type="button"
        value="0927967530245"
      >
        Variant 3 - £97.68- Email When in Stock
      </button>
    </Button>
  </div>
</VariantSelector>
`;
